### PR TITLE
Feature/appc 401 reduce wallet size

### PR DESCRIPTION
**Pull request by Trinkes**

### CODE REVIEW

File: app/src/androidTest/java/com/asf/wallet/WalletRepoTest.java [hunk 1]
```diff
@@ -39,12 +46,14 @@
...
     PreferenceRepositoryType preferenceRepositoryType = new SharedPreferenceRepository(context);
-    AccountKeystoreService accountKeystoreService =
-        new GethKeystoreAccountService(new File(context.getFilesDir(), "store"));
-    EthereumNetworkRepositoryType networkRepository =
-        new EthereumNetworkRepository(preferenceRepositoryType);
+    AccountKeystoreService accountKeystoreService = new Web3jKeystoreAccountService(
+        new KeyStoreFileManager(new File(context.getFilesDir(), "store").getAbsolutePath(),
+            new ObjectMapper()), Schedulers.io(),
+        new ObjectMapper());
+    //EthereumNetworkRepositoryType networkRepository =
```
Comment (JoaoRaimundo) [inline app/src/androidTest/java/com/asf/wallet/WalletRepoTest.java]: Remove if not used.

File: app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  private String getFilePath(String accountAddress, String path) {
+    File file = new File(path);
+    if (!file.isDirectory()) {
+      if (file.getName()
+          .toLowerCase()
+          .contains(accountAddress.toLowerCase())) {
+        return file.getAbsolutePath();
+      }
+    }
+    if (file.isDirectory()) {
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java]: Should we start with the 'isDirectory' if statement and the use 'else' to check if the file name matches the account address?

File: app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      for (File subFile : file.listFiles()) {
+        String filePath = getFilePath(accountAddress, subFile.getPath());
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java]: Can't we just return the recursive call result?

File: app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        getFilePath(removeHexIndicator(accountAddress), keystoreFolderPath),
+        "Wallet with address: " + accountAddress + " not found");
+  }
+
+  private String removeHexIndicator(String accountAddress) {
+    return accountAddress.replace("0x", "");
+  }
+
+  private String getFilePath(String accountAddress, String path) {
+    File file = new File(path);
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java:38]: Should we validate for a possible null 'path'?

File: app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java (hunk 2 above)
Comment (Trinkes) [reply app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java]: corrected

**Pull request merged**